VikingQuiz.Api/Utilities/AutheticationService.cs

```diff
@@ -13,26 +13,31 @@ namespace VikingQuiz.Api.Utilities
 {
     public class AuthenticationService
     {
-        private readonly IConfiguration _config;
         private readonly VikinQuizContext context;
+        private readonly IConfiguration config;
 
         public AuthenticationService(IConfiguration config, VikinQuizContext context)
         {
-            this._config = config;
+            this.config = config;
             this.context = context;
         }
-        public string GenerateTokenForUser(User user)
+
+        /// <param name="role">Possible values: Player / Email / ResetPassword / Admin </param>
+        public string GenerateTokenForUser(User user, int hoursToExpiry = 2, string role = "player")
         {
-            var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_config["Jwt:Key"]));
+            var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(config["Jwt:Key"]));
             var creds = new SigningCredentials(key, SecurityAlgorithms.HmacSha256);
-            var claims = new List<Claim> { new Claim(ClaimTypes.NameIdentifier, user.Id.ToString()) };
-            var token = new JwtSecurityToken(_config["Jwt:Issuer"], _config["Jwt:Issuer"], expires: DateTime.Now.AddHours(2), signingCredentials: creds, claims: claims);
+            var claims = new List<Claim> {
+                new Claim(ClaimTypes.NameIdentifier, user.Id.ToString()),
+                new Claim(ClaimTypes.Role, role)
+            };
+            var token = new JwtSecurityToken(config["Jwt:Issuer"], config["Jwt:Issuer"], expires: DateTime.Now.AddHours(hoursToExpiry), signingCredentials: creds, claims: claims);
             return new JwtSecurityTokenHandler().WriteToken(token);
         }
 
-        public User GetUserByCredentials(string username, string password)
+        internal User GetUserByCredentials(string email, string password)
```

Comment:
Unless you really need to make this method accessible only in the current assembly, leave it public.